make lokinet-bootstrap target toggleable

majestrate opened the pull request:
add cmake option to disable building lokinet-bootstrap binary
addresses #1733

majestrate committed:
make lokinet-bootstrap target toggleable

add cmake option to disable building lokinet-bootstrap binary

jagerman commented on daemon/CMakeLists.txt:
```diff
@@ -1,12 +1,21 @@
+if(APPLE)
+  set(WITH_BOOTSTRAP OFF)
+else()
+  option(WITH_BOOTSTRAP "build lokinet-bootstrap tool" ON)
+endif()
```
```suggestion
set(DEFAULT_WITH_BOOTSTRAP ON)
if(APPLE)
  set(DEFAULT_WITH_BOOTSTRAP OFF)
endif()
option(WITH_BOOTSTRAP "build lokinet-bootstrap tool" ${DEFAULT_WITH_BOOTSTRAP})
```
so that you can still build it on Apple if you explicitly enable it.

jagerman commented on daemon/CMakeLists.txt:
```diff
@@ -1,12 +1,21 @@
   add_executable(lokinet lokinet.cpp)
-  add_executable(lokinet-bootstrap lokinet-bootstrap.cpp)
-  enable_lto(lokinet lokinet-vpn lokinet-bootstrap)
+  if(WITH_BOOTSTRAP)
+    add_executable(lokinet-bootstrap lokinet-bootstrap.cpp)
+    enable_lto(lokinet lokinet-vpn lokinet-bootstrap)
+  else()
+    enable_lto(lokinet lokinet-vpn)
+  endif()
```
Slightly cleaner:
```suggestion
  enable_lto(lokinet lokinet-vpn)

  if(WITH_BOOTSTRAP)
    add_executable(lokinet-bootstrap lokinet-bootstrap.cpp)
    enable_lto(lokinet-bootstrap)
  endif()
```

jagerman committed:
Apply suggestions from code review

jagerman committed:
Add mainnet/testnet bootstrap files into repo

We use them everywhere now (include the debs) so it makes sense to have
them bundled here rather than needing a download during package build
(in particular for debs that's considered a bad thing).

jagerman commented:
Added a commit to push the default mainnet/testnet bootstrap files into the repo, since we're bundling them in all the builds now (they are already in the deb repos; I'll switch those out with these copies in the next release).
